### PR TITLE
Piwigo 

### PR DESCRIPTION
**Pull request by AfroUSApl**
Hello, I would like t oadd my plugin for TrueNAS.
This is plugin to create Piwigo, an open source photo gallery software for the web. Designed for organisations, teams and individuals.
I have manage to test this couple of times on my TrueNAS 12.2, all seems to work well.

This is my repository with files required to get this plugin working under TreeNAS:
https://github.com/AfroUSApl/iocage-plugin-piwigo

Piwigo 11.3.0
Operating system: FreeBSD
System: HP MicroServer Gen8, Xeon E3-1220L V2, 16GB RAM ECC
PHP: 7.3.27 (Show info) [2021-02-13 14:39:51]
MySQL: 5.5.5-10.5.8-MariaDB [2021-02-13 14:39:51]
Graphics Library: External ImageMagick 7.0.10-24

### CODE REVIEW

File: piwigo.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    "name": "Piwigo",
+    "plugin_schema": "2",
+    "release": "11.4-RELEASE",
```
Comment (tprelog) [inline piwigo.json:4]:
```suggestion
    "release": "12.2-RELEASE",
```

**Comment by tprelog** [comment]
Hi @AfroUSApl 

You need to also add your plugin to the [INDEX](https://github.com/ix-plugin-hub/iocage-plugin-index/blob/master/INDEX)

File: piwigo.json (hunk 1 above)
Comment (AfroUSApl) [reply piwigo.json:4]: I will test this now if this works. Give me 10min ;)

**Comment by AfroUSApl** [comment]
> Hi @AfroUSApl
> 
> You need to also add your plugin to the [INDEX](https://github.com/ix-plugin-hub/iocage-plugin-index/blob/master/INDEX)

File: piwigo.json (hunk 1 above)
Comment (AfroUSApl) [reply piwigo.json]: Got it to work without problems, Release 12.2 is working, confirmed ;)

Comment (AfroUSApl) [reply piwigo.json]: `code`

**Comment by tprelog** [comment]
> > Hi @AfroUSApl
> > You need to also add your plugin to the [INDEX](https://github.com/ix-plugin-hub/iocage-plugin-index/blob/master/INDEX)
> 
> Did this just now, please check

Hi - Please include the updated INDEX here and re-open this pull request -- you should have all 3 things together in on request

>After creating the JSON file and adding an icon, add the plugin to the INDEX file. If the plugin is approved, it appears in the list of iocage plugins.

**Comment by AfroUSApl** [comment]
INDEX updated, please check ;)

**Comment by tprelog** [comment]
Looks good :+1: 

I've Piwigo in the past -- I'm looking forward to this one

**Comment by AfroUSApl** [comment]
> Looks good 👍
> 
> I've Piwigo in the past -- I'm looking forward to this one

Thanks, I have over 57k of photos, working very well, happy to have this instead of google photos ;)

**Comment by tprelog** [comment]
> Not sure if after adding all, I need to move files from my repo to the main ix-plugin- ??

Nope, you have everything you need!

**Comment by AfroUSApl** [comment]
> > Not sure if after adding all, I need to move files from my repo to the main ix-plugin- ??
> 
> Nope, you have everything you need!

Thank you ;)

**Comment by tprelog** [comment]
> So the files I have building this plugin will stay on my repo

Yes, err more accurate to say the will download from the repo defined in the manifest -  These are the community plugins so it will not be automatically by ix-systems

but it could be changed to another repo via normal pull request, though this typically won't happen unless you abandon the plugin scripts, they stops working and some else wants to take over and fix it.

**Comment by AfroUSApl** [comment]
> > So the files I have building this plugin will stay on my repo
> 
> Yes, err more accurate to say the will download from the repo defined in the manifest - These are the community plugins so it will not be automatically by ix-systems
> 
> but it could be changed to another repo via normal pull request, though this typically won't happen unless you abandon the plugin scripts, they stops working and some else wants to take over and fix it.

Great, so I will try to keep it monitored, as I think now when I move from google photos to piwigo on my own server.
How much time until this plugin will be available to community?

**Comment by tprelog** [comment]
> How much time until this plugin will be available to community?

Usually not too long. It's a two step process, first to the master branch, which is, I think, considered a short testing period. Then master will get synced to the release branch (currently that's 12.2-RELEASE for TrueNAS) - Once it hit's the release branch, it will available to everyone. It's usually only a few days to a week from time you make pull request, till it's in the release

**Comment by AfroUSApl** [comment]
> > How much time until this plugin will be available to community?
> 
> Usually not too long. It's a two step process, first to the master branch, which is, I think, considered a short testing period. Then master will get synced to the release branch (currently that's 12.2-RELEASE for TrueNAS) - Once it hit's the release branch, it will available to everyone. It's usually only a few days to a week from time you make pull request, till it's in the release

Great, cant wait to see this beeing improved. One more thing, why INDEX of comunity plugins is not sorted alphabetically? Is it OK if I will do it tomorrow? Should I make another pull request?

**Comment by tprelog** [comment]
I agree, it would be nice to see it sorted. Hang off for a minute though.

IMO it should be a separate pull request, however it has sorted in one of the commits for the Movie Night plugin (also has pull request pending) So If that's approved it will already done :+1:  - let's wait and see what happens

**Comment by AfroUSApl** [comment]
I did found a problem with not updating PHP.INI by post_install.sh
Now its fixed, all changes are going through ;)

**Comment by jsegaert** [comment]
I also tested, and it looks good; I put my [comments in the forum thread](https://www.truenas.com/community/threads/piwigo-plugin-for-truenas-ready-to-test.91146/#post-632037).  

And yes, you are supposed to add it to the INDEX in alphabetical order, otherwise it will look odd in the TrueNAS gui.

**Comment by AfroUSApl** [comment]
> I also tested, and it looks good; I put my [comments in the forum thread](https://www.truenas.com/community/threads/piwigo-plugin-for-truenas-ready-to-test.91146/#post-632037).
> 
> And yes, you are supposed to add it to the INDEX in alphabetical order, otherwise it will look odd in the TrueNAS gui.

Thank you for testing!

I did add a line info about host to post_install.sh and add new branch for 12.2_RELEASE in my rep. Thank you for suggestions 😊

**Review by kmoore134 — APPROVED** [review]

**Comment by AfroUSApl** [comment]
Thank you all for helping me to fix all issues. Hope people will be happy to test and use this plugin as a replacement for Google Photos. as I am now :)

Should I close with comment or leavie it like it is?

Regards,
Thomas

**Pull request merged**